library/iFixit/Matryoshka/DeleteBeforeUpdate.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace iFixit\Matryoshka;
+
+use iFixit\Matryoshka;
+
+/**
+ * Ensures that a delete is issued before trying to update an existing key.
+ *
+ * This layer is useful for working with mcrouter. mcrouter provides a reliable
+ * delete stream, but can't guarantee updates are recorded.
+ */
+class DeleteBeforeUpdate extends BackendWrap {
+   public function set($key, $value, $expiration = 0) {
+      if ($this->backend->add($key, $value) === false) {
```

Comment:
Will the `add` call here return `false` for _any_ error?

Reply:
ah good point, it's only a subset of errors we care about

>will return Memcached::RES_NOTSTORED if the key already exists.

It appears there are error codes that can be retrieved

Reply:
I think I'm OK with false here. The chances are slim that another error is returned and the affect is a negligible performance penalty.